src/java.base/share/classes/jdk/internal/vm/Continuation.java

```diff
@@ -0,0 +1,525 @@
+        return stackWalker(options, this.scope);
+    }
+
+    /**
+     * Creates a StackWalker for this continuation and enclosing ones up to the given scope
+     * @param options the StackWalker's configuration options
+     * @param scope the delimiting continuation scope for the stack
+     * @return a new StackWalker
+     */
+    public StackWalker stackWalker(Set<StackWalker.Option> options, ContinuationScope scope) {
+        return JLA.newStackWalkerInstance(options, scope, innermost());
+    }
+
+    /**
+     * Obtains a stack trace for this unmounted continuation
+     * @return the stack trace
+     * @throws IllegalStateException if the continuation is mounted
+     */
+    public StackTraceElement[] getStackTrace() {
+        return stackWalker(EnumSet.of(StackWalker.Option.SHOW_REFLECT_FRAMES))
+            .walk(s -> s.map(StackWalker.StackFrame::toStackTraceElement)
+            .toArray(StackTraceElement[]::new));
+    }
+
+    /// Support for StackWalker
+    public static <R> R wrapWalk(Continuation inner, ContinuationScope scope, Supplier<R> walk) {
+        try {
+            for (Continuation c = inner; c != null && c.scope != scope; c = c.parent)
+                c.mount();
+            return walk.get();
+        } finally {
+            for (Continuation c = inner; c != null && c.scope != scope; c = c.parent)
+                c.unmount();
+        }
+    }
+
+    private Continuation innermost() {
+        Continuation c = this;
+        while (c.child != null)
+            c = c.child;
+        return c;
+    }
+
+    private void mount() {
+        if (!compareAndSetMounted(false, true))
+            throw new IllegalStateException("Mounted!!!!");
+    }
+
+    private void unmount() {
+        setMounted(false);
+    }
+
+    /**
+     * Mounts and runs the continuation body. If suspended, continues it from the last suspend point.
+     */
+    public final void run() {
+        while (true) {
+            mount();
+            JLA.setScopeLocalCache(scopeLocalCache);
+
+            if (done)
+                throw new IllegalStateException("Continuation terminated");
+
+            Thread t = currentCarrierThread();
+            if (parent != null) {
+                if (parent != JLA.getContinuation(t))
+                    throw new IllegalStateException();
+            } else
+                this.parent = JLA.getContinuation(t);
+            JLA.setContinuation(t, this);
+
+            try {
+                boolean isVirtualThread = (scope == JLA.virtualThreadContinuationScope());
+                if (!isStarted()) { // is this the first run? (at this point we know !done)
+                    enterSpecial(this, false, isVirtualThread);
+                } else {
+                    assert !isEmpty();
+                    enterSpecial(this, true, isVirtualThread);
+                }
+            } finally {
+                fence();
+                StackChunk c = tail;
```

Comment:
`c` is not used

Reply:
Ron has done some cleanup so this is removed.